Fix AAD MailNickName attribute value calculation

erdinckorpeoglu-msft opened the pull request:
Changed ordering of the attributes taken into account for Azure AD MailNickName attribute value calculation. In addition to this, modified scenarios as required.

azurecla commented:
Hi __@jastru__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

erkorpeo-msft committed:
Fix AAD MailNickName attribute value calculation

Changed ordering of the attributes taken into account for Azure AD MailNickName attribute value. In addition to this, modified scenarios as required.
